src/resp.rs

```diff
@@ -388,18 +388,30 @@ pub trait ToRespInteger {
     fn to_resp_integer(self) -> RespValue;
 }
 
-macro_rules! integer_into_resp {
-    ($t:ty) => {
-        into_resp!($t, to_resp_integer);
-    };
+impl<T: ToRespInteger> From<T> for RespValue {
+    fn from(value: T) -> Self {
+        value.to_resp_integer()
+    }
 }
 
-impl ToRespInteger for usize {
-    fn to_resp_integer(self) -> RespValue {
-        RespValue::Integer(self as i64)
+macro_rules! to_resp_integer_impls {
+    ($($int:ty,)+) => {
+        $(
+            impl ToRespInteger for $int {
+                fn to_resp_integer(self) -> RespValue {
+                    RespValue::Integer(self as i64)
+                }
+            }
+        )+
     }
 }
-integer_into_resp!(usize);
+
+to_resp_integer_impls! {
+    i8, i16, i32, i64, isize,
+    u8, u16, u32, u64, usize,
+    i128,
+    u128,
+}
```

Comment:
There are a couple of data modelling dilemmas here.

One is that, according to the Redis RESP spec, a Redis integer is a signed 64-bit value (if I'm remembering correctly, I'll have to check); so automatically converting larger integers, particularly `i128` and `u128`, will lead to these values being truncated. It would probably be safer to leave those out and require clients to convert values appropriately instead. (In theory we also have the same problem with `u64` and `usize` - but these are currently supported so need to be retained for backward compatibility, but perhaps should be removed in a future release.)

The second is that we should retain some symmetry converting to a `RespValue` and converting from a `RespValue` to avoid ambiguity when reading we can't automatically assume that a `u8` is an integer value, it might be part of the bulk string (because Redis bulk strings are byte arrays really).

Because of both of these, I think we should only apply this macro to the same integers that we apply the "from" macro to, specifically `i32`, `i64`, `size`, `u32`, `u64` and `size`.
```suggestion
to_resp_integer_impls! {
    i32, i64, isize,
    u32, u64, usize
}
```

Reply:
I'm fine with the first point, while have different opinions on the second.

According to the Redis RESP spec, `Integer`s and `BulkString`s are encoded in different formats.
When converting to a `RespValue`, a `u8` is indeed an integer value, and can not be part of a bulk string.
When reading from a `RespValue`, a byte encoded in `Integer`or `BulkString` format will always clear.

After all, `ToRespInteger` is used to convert something to `RespValue` and has nothing to do with reading from `RespValue`.